core/src/OpenMP/Kokkos_OpenMP_Parallel.hpp

```diff
@@ -210,18 +207,19 @@ class ParallelFor<FunctorType, Kokkos::MDRangePolicy<Traits...>,
 #pragma omp parallel for schedule(static, 1) \
     num_threads(m_instance->thread_pool_size())
     KOKKOS_PRAGMA_IVDEP_IF_ENABLED
-    for (index_type iwork = 0; iwork < m_mdr_policy.m_num_tiles; ++iwork) {
-      iterate_type(m_mdr_policy, m_functor)(iwork);
+    for (index_type iwork = 0; iwork < m_iter.m_rp.m_num_tiles; ++iwork) {
+      m_iter(iwork);
     }
   }
 
  public:
   inline void execute() const {
-    if (execute_in_serial()) {
-      ParallelFor::exec_range(m_mdr_policy, m_functor, 0,
-                              m_mdr_policy.m_num_tiles);
+#ifndef KOKKOS_COMPILER_INTEL
+    if (execute_in_serial(m_iter.m_rp.space())) {
+      exec_range(0, m_iter.m_rp.m_num_tiles);
       return;
     }
+#endif
```

Comment:
Should this `#endif` be moved up before line 221? It looks to me like the braces are not matched within the guards

Reply:
I think you're misreading the diff, because Github is formatting it inconveniently. The entire block `if (foo) { bar }` is enclosed in the `#ifdef / #endif`